discovery/openstack/instance.go

```diff
@@ -146,12 +146,17 @@ func (i *InstanceDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group,
 				openstackLabelUserID:         model.LabelValue(s.UserID),
 			}
 
-			flavorID, ok := s.Flavor["id"].(string)
-			if !ok {
-				level.Warn(i.logger).Log("msg", "Invalid type for flavor id, expected string")
-				continue
+			flavorName, nameOk := s.Flavor["original_name"].(string)
+			if !nameOk {
```

Comment:
I'd keep the ID check at the beginning, but as you like.
Let's add a comment

```suggestion
                        // "original_name" is only available for microversion >= 2.47. It was added in favor of "id".
			if !nameOk {
```